sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/UnionEstimationSuite.scala

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.statsEstimation
+
+import org.apache.spark.sql.catalyst.expressions.{AttributeMap, AttributeReference}
+import org.apache.spark.sql.catalyst.plans.logical
+import org.apache.spark.sql.catalyst.plans.logical.{ColumnStat, Union}
+import org.apache.spark.sql.types.{ByteType, DateType, Decimal, DecimalType, DoubleType, FloatType, IntegerType, LongType, ShortType, TimestampType}
+
+class UnionEstimationSuite extends StatsEstimationTestBase {
+
+  test("test row size estimation") {
+    val attrInt = AttributeReference("cint", IntegerType)()
+
+    val sz = Some(BigInt(1024))
+    val child1 = StatsTestPlan(
+      outputList = Seq(attrInt),
+      rowCount = 2,
+      attributeStats = AttributeMap(Nil),
+      size = sz)
+
+    val child2 = StatsTestPlan(
+      outputList = Seq(attrInt),
+      rowCount = 2,
+      attributeStats = AttributeMap(Nil),
+      size = sz)
+
+    val union = Union(Seq(child1, child2))
+    val expectedStats = logical.Statistics(sizeInBytes = 2 * 1024, rowCount = Some(4))
+    assert(union.stats === expectedStats)
+  }
+
+  test("col stats estimation") {
+    val sz = Some(BigInt(1024))
+
+    val attrInt = AttributeReference("cint", IntegerType)()
+    val attrDouble = AttributeReference("cdouble", DoubleType)()
+    val attrShort = AttributeReference("cshort", ShortType)()
+    val attrLong = AttributeReference("clong", LongType)()
+    val attrByte = AttributeReference("cbyte", ByteType)()
+    val attrFloat = AttributeReference("cfloat", FloatType)()
+    val attrDecimal = AttributeReference("cdecimal", DecimalType(5, 4))()
```

Comment:
Could you add some tests for other precision&scale combinations?

Reply:
Added test.